hack/make.ps1

```diff
@@ -0,0 +1,412 @@
+try {
+
+    Write-Host -ForegroundColor Cyan "INFO: make.ps1 starting at $(Get-Date)"
+
+    if ($All) { $Client=$True; $Daemon=$True; $DCO=$True; $PkgImports=$True; $GoFormat=$True; $TestUnit=$True }
+
+    $root=$(pwd)
+    if ($Binary) { $Client = $True; $Daemon = $True }
+
+    # Make sure we have something to do
+    if (-not($Client) -and -not($Daemon) -and -not($DCO) -and -not($PkgImports) -and -not($GoFormat) -and -not($TestUnit)) { Throw "Nothing to do. Try -All for everything I can do" }
+
+    # Verify git is installed
+    if ($(Get-Command git -ErrorAction SilentlyContinue) -eq $nil) { Throw "Git does not appear to be installed" }
+
+    # Get the git commit. This will also verify if we are in a repo or not.
+    $gitCommit=Get-GitCommit + "$CommitSuffix"
+    if ($CommitSuffix -ne "") { $gitCommit += "-"+$CommitSuffix }
+
+    # Get the version of docker (eg 1.14.0-dev)
+    $dockerVersion=Get-DockerVersion
+
+    # Give a warning if we are not running in a container and are building binaries or running unit tests. 
+    # Not relevant for validation tests as these are fine to run outside of a container.
+    if ($Client -or $Daemon -or $TestUnit) { Check-InContainer }
+
+    # Verify GOPATH is set
+    if ($env:GOPATH.Length -eq 0) { Throw "Missing GOPATH environment variable. See https://golang.org/doc/code.html#GOPATH" }
+
+    # Run autogen if building binaries or running unit tests.
+    if ($Client -or $Daemon -or $TestUnit) {
+        Write-Host "INFO: Invoking autogen..."
+        .\hack\make\.go-autogen.ps1 -CommitString $gitCommit -DockerVersion $dockerVersion
+    }
+
+    # Create the bundles directory if building binaries
+    if (($Client -or $Daemon) -and  -not (Test-Path ".\bundles")) { New-Item ".\bundles" -ItemType Directory | Out-Null }
+
+    # Generate the build tags
+    if ($Client -or $Daemon) {
+        $buildTags="autogen"
+        if ($Noisy) { $verboseParm=" -v" }
+        if ($Race) { 
+            Write-Warning "Using race detector"
+            $raceParm=" -race" 
+        }
+        $allParam=""
+        if ($ForceBuildAll) { $allParm=" -a" }
+        $optParam=""
+        if ($NoOptimise) { $optParm=' -gcflags """ + "-N -l" + """' }
+    }
+
+    # Build the daemon binary
+    if ($Daemon) { Build-Daemon }
+
+    # Build the client binary
+    if ($Client) { Build-Client }
+
+    # We need the head and upstream commits for certain tests
+    if ($DCO -or $PkgImports -or $GoFormat) {
+        $headCommit=Get-HeadCommit
+        $upstreamCommit=Get-UpstreamCommit
+    }
+
+    # Run DCO validation
+    if ($DCO) { Validate-DCO $headCommit $upstreamCommit }
+
+    # Run `gofmt` validation
+    if ($GoFormat) { Validate-GoFormat $headCommit $upstreamCommit }
+
+    # Run pkg isolation validation
+    if ($PkgImports) { Validate-PkgImports $headCommit $upstreamCommit }
+
+    # Run unit tests
+    if ($TestUnit) { Run-UnitTests }
+
+    # Gratuitious ASCII art.
+    if ($Daemon -or $Client) {
```

Comment:
Why only show the 'OK' message when building the daemon or client, but the `Failed` for any failure? It might be nice to see the art when all the unit tests pass. :)